pkg/config/file.go

```diff
@@ -33,42 +36,80 @@ import (
 var _ Loader = (*FileLoader)(nil)
 
 type FileLoader struct {
-	path     string
+	config   LoaderConfig
 	cRuntime chan<- runtime.Config
 	done     chan struct{}
+	fsys     fs.FS
 }
 
 func NewFileLoader(cfg *Config, cRuntime chan<- runtime.Config) *FileLoader {
 	return &FileLoader{
-		path:     cfg.Loader.File.Path,
+		config:   cfg.Loader,
 		cRuntime: cRuntime,
+		done:     make(chan struct{}, 1),
+		fsys:     os.DirFS(filepath.Dir(cfg.Loader.File.Path)),
```

Comment:
from os.DirFS docs:

> The directory dir must not be "".

Please add to the validation function of the config, so that the `cfg.Loader.File.Path` may never be `""`.

Reply:
Added some validation to the `Validate()` function and added a unit test for it.